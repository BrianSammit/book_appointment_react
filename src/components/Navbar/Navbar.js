import React, { Component } from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';

export default class Navbar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      clicked: false,
      loggedInStatus: 'NOT_LOGGED_IN',
    };

    this.handleClick = this.handleClick.bind(this);
    this.handleLogoutClick = this.handleLogoutClick.bind(this);
  }

  componentDidMount() {
    this.checkLoginStatus();
  }

  handleClick() {
    const { clicked } = this.state;
    this.setState({ clicked: !clicked });
  }

  handleLogoutClick() {
    const { handleLogout } = this.props; /* eslint-disable-line */
    axios
      .delete('http://localhost:3001/logout', { withCredentials: true })
      .then(() => handleLogout())
      .catch((error) => {
        throw new Error(error);
      });
  }

  checkLoginStatus() {
    const { loggedInStatus } = this.state;
    axios
      .get('http://localhost:3001/logged_in', { withCredentials: true })
      .then((response) => {
        if (response.data.logged_in && loggedInStatus === 'NOT_LOGGED_IN') {
          this.setState({
            loggedInStatus: 'LOGGED_IN',
            user: response.data.user,
          });
        } else if (!response.data.logged_in && loggedInStatus === 'LOGGED_IN') {
          this.setState({
            loggedInStatus: 'NOT_LOGGED_IN',
          });
        }
      })
      .catch((error) => {
        throw new Error(error);
      });
  }

  render() {
    const { clicked } = this.state;
    const { loggedInStatus } = this.state; /* eslint-disable-line */
    return (
      <div className="navbar">
        <button type="button" className="menu-icon" onClick={this.handleClick}>
          <i className={clicked ? 'fas fa-times' : 'fas fa-bars'} />
        </button>
        <div className={clicked ? 'nav-menu-active' : 'nav-menu'}>
          <Link className="nav-link" to="/skates">
            <img
              alt="logo"
              src="https://res.cloudinary.com/ddjesec95/image/upload/v1632449015/Skate%20brands/logo_pqi497.png"
              className="logo"
            />
          </Link>
          <ul className="links">
            <li>
              <Link to="/skates">SKATES</Link>
            </li>
            <li>
              <Link to="/lifestyle">LIFESTYLE</Link>
            </li>
            <li>
              <Link to="/shop">SHOP</Link>
            </li>
            <li>
              <Link to="/test">TESTS</Link>
            </li>
          </ul>
          <div>
            {loggedInStatus === 'NOT_LOGGED_IN' ? (
              <div className="nav-btn">
                <button type="button" className="nav-btn-links">
                  <Link className="nav-btn-link" to="/" onClick={this.checkLoginStatus()}>
                    LOGIN
                  </Link>
                </button>
                <button type="button" className="nav-btn-links">
                  <Link
                    className="nav-btn-link"
                    to="/registration"
                    onClick={this.checkLoginStatus()}
                  >
                    REGISTER
                  </Link>
                </button>
              </div>
            ) : (
              <div className="nav-btn">
                <button type="button" className="nav-btn-links" onClick={this.checkLoginStatus()}>
                  <Link className="nav-btn-link" to="/" onClick={() => this.handleLogoutClick()}>
                    LOGOUT
                  </Link>
                </button>
              </div>
            )}
          </div>
          <ul className="nav-icons">
            <li className="nav-icons-li">
              <a
                className="nav-icons-link"
                href="https://twitter.com/CruzSammit"
                target="_blank"
                rel="noreferrer"
              >
                <i className="fab fa-twitter" />
              </a>
            </li>
            <li className="nav-icons-li">
              <a
                className="nav-icons-link"
                href="https://www.facebook.com/groups/easkate4"
                target="_blank"
                rel="noreferrer"
              >
                <i className="fab fa-facebook-f" />
              </a>
            </li>
            <li className="nav-icons-li">
              <a
                className="nav-icons-link"
                href="https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign"
                target="_blank"
                rel="noreferrer"
              >
                <i className="fab fa-behance-square" />
              </a>
            </li>
            <li className="nav-icons-li">
              <a
                className="nav-icons-link"
                href="https://github.com/BrianSammit"
                target="_blank"
                rel="noreferrer"
              >
                <i className="fab fa-github" />
              </a>
            </li>
            <li className="nav-icons-li">
              <a
                className="nav-icons-link"
                href="https://www.pinterest.es/uolker/skate/"
                target="_blank"
                rel="noreferrer"
              >
                <i className="fab fa-pinterest-p" />
              </a>
            </li>
          </ul>
          <div className="copy">
            <p>Copyright Â© 2021 Brian Sammit. All rights reserved.</p>
          </div>
        </div>
      </div>
    );
  }
}
