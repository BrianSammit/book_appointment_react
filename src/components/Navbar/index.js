import React from 'react';
import { Nav, NavLink, NavMenu, NavBtn, NavBtnLink } from './NavBar';

const Navbar = () => {
  return (
    <>
      <Nav>
        <NavLink to="/">
          <img
            src="https://res.cloudinary.com/ddjesec95/image/upload/v1632449015/Skate%20brands/logo_pqi497.png"
            className="logo"
          ></img>
        </NavLink>
        <NavMenu>
          <NavLink to="/skates">SKATE</NavLink>
          <NavLink to="/lifestyle">LIFESTYLE</NavLink>
          <NavLink to="/shop">SHOP</NavLink>
          <NavLink to="/test">TEST</NavLink>
        </NavMenu>
        <NavBtn>
          <NavBtnLink to="/signin">Sign In</NavBtnLink>
        </NavBtn>
      </Nav>
    </>
  );
};

export default Navbar;
