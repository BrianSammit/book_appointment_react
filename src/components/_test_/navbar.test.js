/**
 * @jest-environment jsdom
 */

import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Navbar from '../Navbar/Navbar';

describe('Creates a component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Navbar />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders a component', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );
    const element = screen.getByText('SKATES');
    expect(element).toBeInTheDocument();
  });
});
