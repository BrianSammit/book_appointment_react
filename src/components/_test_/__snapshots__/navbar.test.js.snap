// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a component renders correctly 1`] = `
<div
  className="navbar"
>
  <button
    className="menu-icon"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-bars"
    />
  </button>
  <div
    className="nav-menu"
  >
    <a
      className="nav-link"
      href="/skates"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo"
        src="https://res.cloudinary.com/ddjesec95/image/upload/v1632449015/Skate%20brands/logo_pqi497.png"
      />
    </a>
    <ul
      className="links"
    >
      <li>
        <a
          href="/skates"
          onClick={[Function]}
        >
          SKATES
        </a>
      </li>
      <li>
        <a
          href="/lifestyle"
          onClick={[Function]}
        >
          LIFESTYLE
        </a>
      </li>
      <li>
        <a
          href="/shop"
          onClick={[Function]}
        >
          SHOP
        </a>
      </li>
      <li>
        <a
          href="/test"
          onClick={[Function]}
        >
          TESTS
        </a>
      </li>
    </ul>
    <div>
      <div
        className="nav-btn"
      >
        <button
          className="nav-btn-links"
          type="button"
        >
          <a
            className="nav-btn-link"
            href="/"
            onClick={[Function]}
          >
            LOGIN
          </a>
        </button>
        <button
          className="nav-btn-links"
          type="button"
        >
          <a
            className="nav-btn-link"
            href="/registration"
            onClick={[Function]}
          >
            REGISTER
          </a>
        </button>
        <button
          className="nav-btn-links"
          type="button"
        >
          <a
            className="nav-btn-link"
            href="/"
            onClick={[Function]}
          >
            LOGOUT
          </a>
        </button>
      </div>
    </div>
    <ul
      className="nav-icons"
    >
      <li
        className="nav-icons-li"
      >
        <a
          className="nav-icons-link"
          href="https://twitter.com/CruzSammit"
          rel="noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-twitter"
          />
        </a>
      </li>
      <li
        className="nav-icons-li"
      >
        <a
          className="nav-icons-link"
          href="https://www.facebook.com/groups/easkate4"
          rel="noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-facebook-f"
          />
        </a>
      </li>
      <li
        className="nav-icons-li"
      >
        <a
          className="nav-icons-link"
          href="https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign"
          rel="noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-behance-square"
          />
        </a>
      </li>
      <li
        className="nav-icons-li"
      >
        <a
          className="nav-icons-link"
          href="https://github.com/BrianSammit"
          rel="noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-github"
          />
        </a>
      </li>
      <li
        className="nav-icons-li"
      >
        <a
          className="nav-icons-link"
          href="https://www.pinterest.es/uolker/skate/"
          rel="noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-pinterest-p"
          />
        </a>
      </li>
    </ul>
    <div
      className="copy"
    >
      <p>
        Copyright Â© 2021 Brian Sammit. All rights reserved.
      </p>
    </div>
  </div>
</div>
`;
